presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -324,8 +342,28 @@ public int nextInt()
         private final GroupedAccumulator aggregation;
         private AggregationNode.Step step;
         private final int intermediateChannel;
+        private final List<Integer> orderByChannels;
+        private final List<SortOrder> orderings;
+        private final int[] orderByOutputChannels;
+        private final Optional<GroupByHash> groupByHash;
+        private final Optional<PagesIndex> pagesIndex;
+
+        private Aggregator(
+                AccumulatorFactory accumulatorFactory,
+                AggregationNode.Step step)
+        {
+            this(accumulatorFactory, step, Optional.empty(), ImmutableList.of(), ImmutableList.of(), ImmutableList.of(), Optional.empty(), Optional.empty());
+        }
 
-        private Aggregator(AccumulatorFactory accumulatorFactory, AggregationNode.Step step, Optional<Integer> overwriteIntermediateChannel)
+        private Aggregator(
```

Comment:
I don't think this class has the right abstraction level. It already gets `groupIds` on the call to `addPage`, so it seems weird for it to need `groupByHash`, too.